Content.Server/Body/Systems/BodyReassembleSystem.cs

```diff
@@ -1,229 +1,279 @@
+    /// </summary>
+    private void AddReassembleVerbs(EntityUid uid, BodyReassembleComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
 
-            if (!TryComp<MindComponent>(uid, out var mind) ||
-                !mind.HasMind ||
-                component.CancelToken != null)
-                return;
+        if (!TryComp<MindComponent>(uid, out var mind) ||
+            !mind.HasMind ||
+            component.CancelToken != null)
+            return;
 
-            // doubles the time if you reconstruct yourself
-            var multiplier = args.User == uid ? SelfReassembleMultiplier : 1f;
+        // doubles the time if you reconstruct yourself
+        var multiplier = args.User == uid ? SelfReassembleMultiplier : 1f;
 
-            // Custom verb
-            AlternativeVerb custom = new()
+        // Custom verb
+        AlternativeVerb custom = new()
+        {
+            Text = Loc.GetString("reassemble-action"),
+            Act = () =>
             {
-                Text = Loc.GetString("reassemble-action"),
-                Act = () =>
-                {
-                    StartReassembly(uid, component, multiplier);
-                },
-                IconEntity = uid,
-                Priority = 1
-            };
-            args.Verbs.Add(custom);
-        }
+                StartReassembly(uid, component, multiplier);
+            },
+            IconEntity = uid,
+            Priority = 1
+        };
+        args.Verbs.Add(custom);
+    }
 
-        private bool GetNearbyParts(EntityUid uid, BodyReassembleComponent component, out HashSet<EntityUid>? partList)
+    private bool GetNearbyParts(EntityUid uid, BodyReassembleComponent component, out HashSet<EntityUid>? partList)
+    {
+        partList = new HashSet<EntityUid>();
+        
+        if (component.BodyParts == null)
+            return false;
+
+        var tempList = new HashSet<EntityUid>(component.BodyParts);
+        tempList.Remove(component.Owner);
+        partList.Add(component.Owner); //the query dislikes the owner
+
+        var entq = GetEntityQuery<BodyPartComponent>();
+        EntityUid? foundPart = null;
+        foreach (var entity in _lookup.GetEntitiesInRange(component.Owner, 2f))
         {
-            partList = new HashSet<EntityUid>();
+            if (!entq.HasComponent(entity))
+                continue;
 
-            if (component.BodyParts == null)
-                return false;
+            foreach (var part in tempList)
+            {
+                var partmeta = MetaData(part);
+                var entmeta = MetaData(entity);
 
-            // Ensures all of the old body part pieces are there
-            var xformQuery = GetEntityQuery<TransformComponent>();
-            var bodyXform = xformQuery.GetComponent(uid);
+                if (partmeta.EntityPrototype == null || entmeta.EntityPrototype == null)
+                    continue;
 
```

Comment:
this shouldn't depend on entity prototypes.
It should probably just look for specific body parts with some sort of limb reassembly component